test/v2/connectors/AmendForeignPropertyPeriodSummaryConnectorSpec.scala

```diff
@@ -78,33 +77,57 @@ class AmendForeignPropertyPeriodSummaryConnectorSpec extends ConnectorSpec {
     foreignNonFhlProperty = Some(Seq(foreignNonFhlPropertyEntry))
   )
 
-  private val request: AmendForeignPropertyPeriodSummaryRequest = AmendForeignPropertyPeriodSummaryRequest(
-    nino = Nino(nino),
-    businessId = businessId,
-    taxYear = TaxYear.fromMtd(taxYear),
-    submissionId = submissionId,
-    body = requestBody
-  )
-
   trait Test {
     _: ConnectorTest =>
-    val connector: AmendForeignPropertyPeriodSummaryConnector = new AmendForeignPropertyPeriodSummaryConnector(
+    def taxYear: TaxYear
+
+    protected val connector: AmendForeignPropertyPeriodSummaryConnector = new AmendForeignPropertyPeriodSummaryConnector(
       http = mockHttpClient,
       appConfig = mockAppConfig
     )
+
+    protected val request: AmendForeignPropertyPeriodSummaryRequest = AmendForeignPropertyPeriodSummaryRequest(
+      nino = Nino(nino),
+      businessId = businessId,
+      taxYear = taxYear,
+      submissionId = submissionId,
+      body = requestBody
+    )
   }
 
   "AmendForeignPropertyPeriodSummaryConnector" must {
     "send a request and return 204 no content" in new IfsTest with Test {
+      def taxYear: TaxYear = TaxYear.fromMtd("2019-20")
```

Comment:
Can we refactor the tests to reuse the outcome? Something like
`"AmendForeignPropertyPeriodSummaryConnector" when {
  val outcome = ...
  "amendForeignPropertyPeriodSummary is called" must {}
  "amendForeignPropertyPeriodSummary is called with a TYS tax year" must {}
}`